backend/src/routes/group.ts

```diff
@@ -0,0 +1,146 @@
+import Router from '@koa/router';
+import Group from '../models/group';
+import User from '../models/user';
+
+const router = new Router({
+	prefix: '/groups',
+});
+
+router.get('/', async (ctx, next) => {
+	ctx.body = await Group.find();
+});
+
+// Finds group by handle.
+router.get('/:grouphandle', async (ctx, next) => {
+	const group = await Group.findOneBy({
+		handle: ctx.params.grouphandle,
+	});
+	console.log(group);
+	if (group === null) {
+		ctx.status = 404;
+		return;
+	}
+	ctx.body = group;
+});
+
+//Create a new group with required group handle and display name.
+router.put('/:grouphandle/:displayname/:creatorhandle', async (ctx, next) => {
```

Comment:
see my other comment about route params vs. request body. here, we can have just grouphandle in the route params, and grab displayname from the request body and creator handle from auth